File: android-bluetooth/src/main/java/com/lightningkite/kwift/bluetooth/BleConnection.shared.kt
package com.lightningkite.kwift.bluetooth

import com.lightningkite.kwift.bytes.Data
import io.reactivex.Observable
import io.reactivex.Single

interface BleConnection {
    val deviceInfo: BleDeviceInfo
    val mtu: Int
    fun readRssi(): Single<Int>
    fun read(characteristic: BleCharacteristic): Single<Data>
    fun write(characteristic: BleCharacteristic, value: Data): Single<Data>
    fun notify(characteristic: BleCharacteristic): Observable<Data>
    fun indicate(characteristic: BleCharacteristic): Observable<Data>
    fun read(descriptor: BleDescriptor): Single<Data>
    fun write(descriptor: BleDescriptor, value: Data): Single<Data>
}

File: ios-bluetooth/KhrysalisBluetooth/com/lightningkite/khrysalis/bluetooth/BleConnection.shared.swift
//Package: com.lightningkite.khrysalis.bluetooth
//Converted using Khrysalis2

import Foundation
import RxSwift
import RxRelay
import Khrysalis



public protocol BleConnection {
    
    var deviceInfo: BleDeviceInfo { get }
    
    var mtu: Int32 { get }
    
    func readRssi() -> Single<Int32>
    
    func read(characteristic: BleCharacteristic) -> Single<Data>
    func read(_ characteristic: BleCharacteristic) -> Single<Data>
    
    func write(characteristic: BleCharacteristic, value: Data) -> Single<Data>
    func write(_ characteristic: BleCharacteristic, _ value: Data) -> Single<Data>
    
    func notify(characteristic: BleCharacteristic) -> Observable<Data>
    func notify(_ characteristic: BleCharacteristic) -> Observable<Data>
    
    func indicate(characteristic: BleCharacteristic) -> Observable<Data>
    func indicate(_ characteristic: BleCharacteristic) -> Observable<Data>
    
    func read(descriptor: BleDescriptor) -> Single<Data>
    func read(_ descriptor: BleDescriptor) -> Single<Data>
    
    func write(descriptor: BleDescriptor, value: Data) -> Single<Data>
    func write(_ descriptor: BleDescriptor, _ value: Data) -> Single<Data>
}
 
 
